server/api/category_controller.py

```diff
@@ -0,0 +1,64 @@
+from flask import Blueprint, request, jsonify
+from models import db, Receipt
+import datetime as dt
+import calendar
+
+category_controller = Blueprint('category_controller',
+                                __name__, url_prefix='/categories')
+
+
+@category_controller.route('/<int:year>/<int:month>', methods=['GET'])
+def get_category_expenses(year, month):
+    num_days = calendar.monthrange(year, month)[1]
+    start_date = dt.date(year, month, 1)
+    end_date = dt.date(year, month, num_days)
+    category_expenses = []
+    category = ''
+    expense = 0
+
+    receipts = (Receipt.query
+                .filter(Receipt.receipt_date <= end_date)
+                .filter(Receipt.receipt_date >= start_date)
+                .order_by(Receipt.category.desc()))
```

Comment:
you should take a look at the SQL aggregate func :D https://www.postgresql.org/docs/9.5/tutorial-agg.html

this will probably be a lot more efficient and much less lines of code

Reply:
it'll probably be along the lines of 

```
SELECT category, SUM(amount)
FROM Receipt
GROUP BY category
ORDER BY SUM(amount)
```

written in the python query way 

if possible, you probably won't even need to manually calculate it :)

Reply:
great! I'll go take a look at it later!